unipop-elastic/src/org/unipop/elastic/document/DocumentController.java

```diff
@@ -135,48 +178,69 @@ public Vertex addVertex(AddVertexQuery uniQuery) {
-            if (document!= null) {
-                IndexRequestBuilder indexRequest = client.prepareIndex(document.getIndex(), document.getType(), document.getId())
-                        .setSource(document.getFields()).setCreate(create);
-                dirty = true;
-                return indexRequest.execute().actionGet();
+            if (document != null) {
+                if (create) {
+                    Get getRequest = new Get.Builder(document.getIndex(), document.getId()).build();
+                    if (this.client.execute(getRequest).isSucceeded()) {
+                        throw new DocumentAlreadyExistsException(
+                                // shard id not interesting.
+                                new ShardId(document.getIndex(), -1),
+                                document.getType(),
+                                document.getId());
+                    }
+                }
+
+                Index index = new Index.Builder(Seq.seq(document.getFields()))
+                        .id(document.getId())
+                        .index(document.getIndex())
+                        .type(document.getType())
+                        .refresh(true)
+                        .build();
+
+//                IndexRequestBuilder indexRequest = client.prepareIndex(document.getIndex(), document.getType(), document.getId())
+//                        .setSource(document.getFields()).setCreate(create);
+                return this.client.execute(index);
             }
         }
         return null;
     }
 
-    private <E extends Element> DeleteResponse delete(Set<? extends DocSchema<E>> schemas, E element) {
+    private <E extends Element> JestResult delete(Set<? extends DocSchema<E>> schemas, E element) throws IOException {
         for(DocSchema<E> schema : schemas) {
             DocSchema.Document document = schema.toDocument(element);
             if (document != null) {
-                DeleteRequestBuilder deleteRequestBuilder = client.prepareDelete(document.getIndex(), document.getType(), document.getId());
+                Delete delete = new Delete.Builder(document.getId())
+                        .index(document.getIndex())
+                        .type(document.getType())
+                        .build();
+
                 dirty = true;
-                return deleteRequestBuilder.execute().actionGet();
+                return this.client.execute(delete);
             }
         }
         return null;
     }
 
-    public void refresh(String... indices) {
-        if (dirty) {
-            //client.admin().indices().prepareRefresh(indices).execute().actionGet();
-            client.admin().indices().prepareRefresh().execute().actionGet();
-            dirty = false;
+    private void refresh(String... indices) throws IOException {
+        if (this.dirty) {
+            Refresh refresh = new Refresh.Builder().addIndex(Seq.of(indices).toList()).build();
```

Comment:
For now ignore the `indices` parameter, because most of the time it's not correct .. 
Did you go over my previous comment?